test/PillarStaking.spec.js

```diff
@@ -0,0 +1,617 @@
+    wethToken = await WETHToken.deploy();
+
+    // deploy DummyPillarToken contract
+    const PillarToken = await ethers.getContractFactory("DummyPillarToken");
+    plrToken = await PillarToken.deploy();
+
+    // deploy PillarStaking contract
+    const PillarStaking = await ethers.getContractFactory("PillarStaking");
+    plrStaking = await PillarStaking.deploy(
+      plrToken.address,
+      wethToken.address,
+      0
+    );
+
+    //transfer dPLR & dWETH tokens to accounts
+    await plrToken
+      .connect(owner)
+      .transfer(addr1.address, ethers.utils.parseEther("1000000"));
+    await plrToken
+      .connect(owner)
+      .transfer(addr2.address, ethers.utils.parseEther("1000000"));
+    await plrToken
+      .connect(owner)
+      .approve(plrStaking.address, ethers.utils.parseEther("10000000"));
+    await plrToken
+      .connect(addr1)
+      .approve(plrStaking.address, ethers.utils.parseEther("1000000"));
+    await plrToken
+      .connect(addr2)
+      .approve(plrStaking.address, ethers.utils.parseEther("1000000"));
+  });
+
+  // DEPLOYMENT //
+  describe("Deployment", () => {
+    it("Deploys Pillar Token without errors", async () => {
+      const ptName = await plrToken.name();
+      const ptSymbol = await plrToken.symbol();
+      const ptTotalSupply = await plrToken.totalSupply();
+      expect(ptName).to.equal("DummyPillarToken");
+      expect(ptSymbol).to.equal("dPLR");
+      expect(ptTotalSupply).to.equal(ethers.utils.parseEther("1000000000"));
+    });
+
+    it("Deploys WETH Token without errors", async () => {
+      const wethName = await wethToken.name();
+      const wethSymbol = await wethToken.symbol();
+      const wethTotalSupply = await wethToken.totalSupply();
+      expect(wethName).to.equal("DummyWETHToken");
+      expect(wethSymbol).to.equal("dWETH");
+      expect(wethTotalSupply).to.equal(ethers.utils.parseEther("1000000000"));
+    });
+
+    it("Deploys Pillar Staking contract without errors", async () => {
+      const psTokenAddress = await plrStaking.stakingToken();
+      const psTokenMinStake = await plrStaking.minStake();
+      const psTokenMaxStake = await plrStaking.maxStake();
+      const psTokenMaxTotalStake = await plrStaking.maxTotalStake();
+      const psStakingState = await plrStaking.getContractState();
+      const psStakingToken = await plrStaking.stakingToken();
+      const psRewardToken = await plrStaking.rewardToken();
+      expect(psTokenAddress).to.equal(plrToken.address);
+      expect(psTokenMinStake).to.equal(ethers.utils.parseEther("10000"));
+      expect(psTokenMaxStake).to.equal(ethers.utils.parseEther("250000"));
+      expect(psTokenMaxTotalStake).to.equal(ethers.utils.parseEther("7200000"));
+      expect(psStakingState).to.equal(0);
+      expect(psStakingToken).to.equal(plrToken.address);
+      expect(psRewardToken).to.equal(wethToken.address);
+    });
+  });
+
+  describe("Staking PLR", () => {
```

Comment:
Add a specific condition to test the same user staking multiple times during the staking period.

Reply:
Done